File: 1-intro/4-exploring-the-waters/js/areSimilar.js
const solution = (a, b) => {
  let aux = [];

  for (let i = 0; i < a.length; i++) {
    if (a[i] != b[i]) {
      aux.push(i);
    }
  }

  if (aux.length == 0) {
    return true;
  }

  if (aux.length == 2) {
    if ((a[aux[1]] == b[aux[0]]) && (a[aux[0]] == b[aux[1]])) {
      return true;
    }
  }

  return false;
}

/* -------------------------------------------------------------------------- */

const solution2 = (a, b) => {
  const a2 = a.filter((val, idx) => val != b[idx]);
  const b2 = b.filter((val, idx) => val != a[idx]);

  return a2.length == 0 || (a2.length == 2 && a2.join(' ') == b2.reverse().join(' '));
}

File: 1-intro/4-exploring-the-waters/cpp/areSimilar.cpp
bool solution(vector<int> a, vector<int> b) {
    vector<int> aux;

    for (int i = 0; i < a.size(); i++) {
        if (a[i] != b[i]) {
            aux.push_back(i);
        }
    }

    if (aux.size() == 0) {
        return true;
    }

    if (aux.size() == 2) {
        if ((a[aux[1]] == b[aux[0]]) and (a[aux[0]] == b[aux[1]])) {
            return true;
        }
    }

    return false;
}
